helix-term/src/application.rs

```diff
@@ -985,7 +997,7 @@ impl Application {
                     MethodCall::ApplyWorkspaceEdit(params) => {
                         apply_workspace_edit(
                             &mut self.editor,
-                            helix_lsp::OffsetEncoding::Utf8,
+                            language_server.offset_encoding(),
```

Comment:
```suggestion
                            offset_encoding,
```

Reply:
Just a small note: This is an existing bug in helix and something I missed in #5894 entirely orthogonal to this PR (that said the fix is rather simple and correct although @poliorcetics suggestions should definitely be applied)